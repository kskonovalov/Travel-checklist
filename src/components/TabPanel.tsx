import { Box } from '@material-ui/core';
import React from 'react';

interface TabPanelProps {
  children?: React.ReactNode;
  index: string | number;
  value: string | number;
  other?: any;
}

const TabPanel: React.FC<TabPanelProps> = ({
  children,
  value,
  index,
  ...other
}) => {
  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
    >
      {value === index && <Box maxWidth="sm">{children}</Box>}
    </div>
  );
};

export default TabPanel;
