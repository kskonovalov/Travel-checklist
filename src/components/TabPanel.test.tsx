import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import TabPanel from './TabPanel';

configure({ adapter: new Adapter() });

const setUp = () => {
  return <TabPanel index="" value="" />;
};

describe('Component: Loader', () => {
  test('Should match snapshot', () => {
    const component = shallow(setUp());
    expect(component).toMatchSnapshot();
  });
});
