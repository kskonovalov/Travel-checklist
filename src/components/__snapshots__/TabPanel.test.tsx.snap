// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader Should match snapshot 1`] = `
<div
  aria-labelledby="vertical-tab-"
  hidden={false}
  id="vertical-tabpanel-"
  role="tabpanel"
>
  <Styled(MuiBox)
    maxWidth="sm"
  />
</div>
`;
